File: backend/Nexus-IMS/src/main/java/com/learn/nexus_ims/services/ChainService.java
package com.learn.nexus_ims.services;

import java.util.List;

import com.learn.nexus_ims.dtos.ChainDto;

public interface ChainService {
    ChainDto createChain(ChainDto chainDto);
    ChainDto updateChain(Long chainId, ChainDto chainDto);
    void deleteChain(Long chainId);
    List<ChainDto> getAllChains();
    ChainDto getChainById(Long chainId);
}

File: frontend/src/services/chainService.js
import api from './api';

export const chainService = {
  getAllChains: async () => {
    const response = await api.get('/chains');
    return response.data;
  },

  getChainById: async (id) => {
    const response = await api.get(`/chains/${id}`);
    return response.data;
  },

  createChain: async (chainData) => {
    const response = await api.post('/chains', chainData);
    return response.data;
  },

  updateChain: async (id, chainData) => {
    const response = await api.put(`/chains/${id}`, chainData);
    return response.data;
  },

  deleteChain: async (id) => {
    await api.delete(`/chains/${id}`);
  },
};
